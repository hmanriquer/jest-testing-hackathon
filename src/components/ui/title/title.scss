.heading-title {
  display: flex;
  padding: 1.5rem;
  justify-content: space-between;
  width: 100%;
}

.heading-title__title {
  font-size: 1.875rem;
}

.heading-title__subtitle {
  font-size: 1.25rem;
  color: #666;
}
