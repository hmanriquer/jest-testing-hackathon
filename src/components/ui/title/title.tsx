import React from 'react';
import './title.scss';

interface TitleProps {
  title: string;
  subtitle?: string;
}

export const Title = ({ title, subtitle }: TitleProps) => {
  return (
    <div className="heading-title">
      <h1 className="heading-title__title">{title}</h1>
      {subtitle && <p className="heading-title__subtitle">{subtitle}</p>}
    </div>
  );
};
