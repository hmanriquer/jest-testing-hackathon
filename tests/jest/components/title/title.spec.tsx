import React from 'react';
import renderer from 'react-test-renderer';
import { Title } from '@components/ui/title/title';

describe('Title Component', () => {
  it('should render with title only', () => {
    const component = renderer.create(<Title title="Hello World" />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with title and subtitle', () => {
    const component = renderer.create(<Title title="Hello World" subtitle="Welcome to the application" />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render the title text correctly', () => {
    const testTitle = 'Test Title';
    const component = renderer.create(<Title title={testTitle} />);

    // Get the rendered output as string and check if it contains the title
    const renderedOutput = JSON.stringify(component.toJSON());
    expect(renderedOutput).toContain(testTitle);
  });

  it('should not render subtitle when not provided', () => {
    const component = renderer.create(<Title title="Hello World" />);

    // Get the rendered output as string and check that it doesn't contain subtitle class
    const renderedOutput = JSON.stringify(component.toJSON());
    expect(renderedOutput).not.toContain('heading-title__subtitle');
  });
});
