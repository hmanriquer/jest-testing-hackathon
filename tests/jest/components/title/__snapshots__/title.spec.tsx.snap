// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Component should render with title and subtitle 1`] = `
<div
  className="heading-title"
>
  <h1
    className="heading-title__title"
  >
    Hello World
  </h1>
  <p
    className="heading-title__subtitle"
  >
    Welcome to the application
  </p>
</div>
`;

exports[`Title Component should render with title only 1`] = `
<div
  className="heading-title"
>
  <h1
    className="heading-title__title"
  >
    Hello World
  </h1>
</div>
`;
